src/Client.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Utopia\Fetch;
+
+/**
+ * Client class
+ * @package Utopia\Fetch
+ */
+class Client
+{
+    public const METHOD_GET = 'GET';
+    public const METHOD_POST = 'POST';
+    public const METHOD_PUT = 'PUT';
+    public const METHOD_PATCH = 'PATCH';
+    public const METHOD_DELETE = 'DELETE';
+    public const METHOD_HEAD = 'HEAD';
+    public const METHOD_OPTIONS = 'OPTIONS';
+    public const METHOD_CONNECT = 'CONNECT';
+    public const METHOD_TRACE = 'TRACE';
+    /**
+     * Flatten request body array to PHP multiple format
+     *
+     * @param array<mixed> $data
+     * @param string $prefix
+     * @return array<mixed>
+     */
+    private function flatten(array $data, string $prefix = ''): array
+    {
+        $output = [];
+        foreach ($data as $key => $value) {
+            $finalKey = $prefix ? "{$prefix}[{$key}]" : $key;
+
+            if (is_array($value)) {
+                $output += $this->flatten($value, $finalKey); // @todo: handle name collision here if needed
+            } else {
+                $output[$finalKey] = $value;
+            }
+        }
+
+        return $output;
+    }
+    /**
+     * This method is used to make a request to the server
+     * @param string $url
+     * @param array<string, string> $headers
+     * @param string $method
+     * @param array<string>|array<string, mixed> $body
+     * @param array<string, mixed> $query
+     * @return Response
+     */
+    private function buildResponse(
+        string $url,
+        array $headers,
+        string $method,
+        array $body,
+        array $query
+    ): Response {
+        // Process the data before making the request
+        if(!$method) { // if method is not set, set it to GET by default
+            $method = self::METHOD_GET;
+        } else { // else convert the method to uppercase
+            $method = strtoupper($method);
+        }
+        // if there are no headers but a body set header to json by default
+        if(!isset($headers['content-type']) && is_array($body) && count($body) > 0) {
+            $headers['content-type'] = 'application/json';
+        }
```

Comment:
Not sure the library should make assumptions about the data... 

Ideally we don't add any magic headers that will leave the user wondering how it worked or did not work... All headers should be explicitly set.

To validate / invalidate my concern I would like you to check the behaviour of the `fetch` library in node and we can circle back on this

Reply:
I checked the `fetch` library behaviour, it doesn't add any default header content-type if not passed. I am removing it from here too.